pkgs/tools/misc/supergfxctl/default.nix

```diff
@@ -0,0 +1,43 @@
+{ lib, rustPlatform, fetchFromGitLab, pkg-config, udev, kmod }:
+
+rustPlatform.buildRustPackage rec {
+  pname = "supergfxctl";
+  version = "2.0.5";
+
+  src = fetchFromGitLab {
+    owner = "asus-linux";
+    repo = pname;
+    rev = version;
+    sha256 = "0qb7ysvbdk0jl26z5z887wf6zq0bsz3991hk2jbp4cxbifd2j8ph";
+  };
+
+  patches = [
+    ./no-config-write.patch
+  ];
+
+  postPatch = ''
+    substituteInPlace data/supergfxd.service \
+      --replace /usr/bin $out/bin
+
+    substituteInPlace src/controller.rs \
+      --replace \"modprobe\" \"${kmod}/bin/modprobe\" \
+      --replace \"rmmod\" \"${kmod}/bin/rmmod\"
+  '';
+
+  nativeBuildInputs = [ pkg-config ];
+  buildInputs = [ udev ];
+
+  cargoHash = "sha256-qT6XRcfx9Zu0PnBxeTeqKv+PiX6tXJpJBdhSr7TjtXk=";
+
+  makeFlags = [ "prefix=${placeholder "out"}" ];
+  # Use default phases since the build scripts install systemd services and udev rules too
+  buildPhase = "buildPhase";
+  installPhase = "installPhase";
```

Comment:
```suggestion
```
Is that not working?

Reply:
No, because then cargo build is run instead of make.

Reply:
We definitely want to run our cargo build phase because it is very different from standard cargo build process. We can run anything else in post phases.
Also we already handle stripping outself and vendoring.

Reply:
This still needs to be fixed properly